backend/controllers/users.controller.js

```diff
@@ -0,0 +1,151 @@
+import User from '../models/users.model.js';
+import likedVideos from '../models/likedvideos.model.js'
+import historyVideos from '../models/history.model.js'
+import Playlist from '../models/playlists.model.js'
+import { mail } from '../utils/mail.js';
+import { generateOTP } from '../utils/generateOTP.js';
+import { OTPMail } from '../utils/sendOTPMail.js';
+import { generateJWToken } from '../utils/generateToken.js';
+import gravatar from 'gravatar';
+import normalize from 'normalize-url';
+import _ from 'lodash';
+import WatchLater from '../models/watchlater.model.js';
+
+//@route   POST api/users/register
+//@desc    Register new User
+//@access  Public
+
+const register = async (req, res) => {
+    try {
+        const { name, email } = req.body;
+
+        if (!name || !email) {
+            return res.status(400).json({ success: false, message: 'name and email are required fields' })
+        }
+
+        const userExists = await User.findOne({ email })
+
+        if (userExists) {
+            return res.status(400).json({ success: false, message: 'Email already in use, Try different email' })
+        }
+        const avatar = normalize(
+            gravatar.url(email, {
+                s: '200',
+                r: 'pg',
+                d: 'mm'
+            }),
+            { forceHttps: true }
+        );
+
+        const user = await new User({
+            name,
+            email,
+            avatar
+        })
+
+        await user.save()
+        await mail(name, email)
+
+        const watchLater = new WatchLater({
+            user: user._id,
+            watchlater: []
+        })
+
+        const LikedVideos = new likedVideos({
+            user: user._id,
+            likedvideos: []
+        })
+
+        const HistoryVideos = new historyVideos({
+            user: user._id,
+            historyvideos: []
+        })
```

Comment:
Use camel casing on line nos 51, 56 and 61.
If you observe line 49 when compared to lines 54 and 59, it must be renamed to `WatchLater`

Reply:
Yes, there is inconsistency in names of my variables but here changing model name again would result in changing in all there controllers files also. So I'll keep this suggestion would not repeat same mistake again.
Hope that's okay for now :)